derp/derphttp/derphttp_client.go

```diff
@@ -644,7 +660,18 @@ func (c *Client) dialNode(ctx context.Context, n *tailcfg.DERPNode) (net.Conn, e
 			}
 			port := "443"
 			if n.DERPPort != 0 {
+				// Question: should the port set in
+				// the tailcfg take precedence or
+				// should the debug envknob? Currently
+				// the tailcfg takes precedence but
+				// that's probably backwards.
 				port = fmt.Sprint(n.DERPPort)
+			} else if envknob.Bool("TS_DEBUG_DERP_USE_HTTP") {
```

Comment:
May want to use envknob.RegisterBool both because it's more efficient to read and because you avoid having to repeat the knob name 3 times.

Reply:
Great, will do.

Reply:
It's now down to two instances so I left these as-is, but I registered the other two envknobs.